src/screens/add_non_hw_screen.tsx

```diff
@@ -0,0 +1,73 @@
+import { NativeStackScreenProps } from "@react-navigation/native-stack";
+import { Box, Button, Center, FormControl, Heading, Input, VStack, TextArea, View} from "native-base";
+import React from "react";
+import { useState } from "react";
+import { SideBarList } from "../components/main_stack";
+import { Platform } from "react-native";
+import DateTimePickerModal from 'react-native-modal-datetime-picker';
+import { Text } from "react-native-svg";
+
+type AddNonHWScreenProps = NativeStackScreenProps<SideBarList, 'AddNonHW'>
+
+
+function AddNonHWScreen({navigation}:AddNonHWScreenProps) {
+  const [selectedDate, setSelectedDate] = useState(new Date('2022-05-31'));
+  const [isDatePickerVisible, setDatePickerVisible] = useState(false);
+
+  const showDatePicker = () => {
+    setDatePickerVisible(true);
+  };
+
+  const hideDatePicker = () => {
+    setDatePickerVisible(false);
+  };
+  const handleConfirm = (date: Date) => {
+    setSelectedDate(date);
+    hideDatePicker();
+  };
+
+  const [textAreaValue, setTextAreaValue] = useState("");
+
+  const demoValueControlledTextArea = e => {setTextAreaValue(e.currentTarget.value)};

```

Comment:
Can you also type this out by replacing this line with `const demoValueControlledTextArea = (e: NativeSyntheticEvent<TextInputChangeEventData>) => {setTextAreaValue(e.nativeEvent.text)};`? I think it is the function typed out, but you might want to test it.